src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ILocalFunctionStatement.cs

```diff
@@ -0,0 +1,309 @@
+        Right: ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1) (Syntax: 'x++')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<LocalFunctionStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void TestLocalFunction_ContainingLocalFunctionParameterReference()
+        {
+            string source = @"
+class C
+{
+    public void M()
+    {
+        int LocalOuter (int x)
+        {
+            /*<bind>*/int Local(int y) => x + y;/*</bind>*/
+            return Local(x);
+        }
+
+        LocalOuter(0);
+    }
+}
+";
+            string expectedOperationTree = @"
+ILocalFunctionStatement (Local Function: System.Int32 Local(System.Int32 y)) (OperationKind.LocalFunctionStatement) (Syntax: 'int Local(i ... ) => x + y;')
+  IBlockStatement (1 statements) (OperationKind.BlockStatement) (Syntax: '=> x + y')
+    IReturnStatement (OperationKind.ReturnStatement) (Syntax: 'x + y')
+      IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32) (Syntax: 'x + y')
+        Left: IParameterReferenceExpression: x (OperationKind.ParameterReferenceExpression, Type: System.Int32) (Syntax: 'x')
+        Right: IParameterReferenceExpression: y (OperationKind.ParameterReferenceExpression, Type: System.Int32) (Syntax: 'y')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<LocalFunctionStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void TestLocalFunction_LocalFunctionReference()
+        {
+            string source = @"
+class C
+{
+    public void M()
+    {
+        int x;
+        int Local(int p1) => x++;
+        int Local2(int p1) => Local(p1);
+        /*<bind>*/int Local3(int p1) => x + Local2(p1);/*</bind>*/
+
+        Local3(x = 0);
+    }
+}
+";
+            string expectedOperationTree = @"
+ILocalFunctionStatement (Local Function: System.Int32 Local3(System.Int32 p1)) (OperationKind.LocalFunctionStatement) (Syntax: 'int Local3( ... Local2(p1);')
+  IBlockStatement (1 statements) (OperationKind.BlockStatement) (Syntax: '=> x + Local2(p1)')
+    IReturnStatement (OperationKind.ReturnStatement) (Syntax: 'x + Local2(p1)')
+      IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32) (Syntax: 'x + Local2(p1)')
+        Left: ILocalReferenceExpression: x (OperationKind.LocalReferenceExpression, Type: System.Int32) (Syntax: 'x')
+        Right: IInvocationExpression (static System.Int32 Local2(System.Int32 p1)) (OperationKind.InvocationExpression, Type: System.Int32) (Syntax: 'Local2(p1)')
```

Comment:
Aleksey, this is due to the dumper printing out "static" when operation.Instance is null. See here: [http://source.roslyn.io/#Roslyn.Test.Utilities/Compilation/OperationTreeVerifier.cs,539](http://source.roslyn.io/#Roslyn.Test.Utilities/Compilation/OperationTreeVerifier.cs,539)

---
In reply to: [121589587](https://github.com/dotnet/roslyn/pull/20177#discussion_r121589587) [](ancestors = 121589587)